device/cuda/src/clusterization/clusterization_algorithm.cu

```diff
@@ -42,19 +48,37 @@ __global__ void ccl_kernel(
     device::ccl_kernel(threadIdx.x, blockDim.x, blockIdx.x, cells_view,
                        modules_view, max_cells_per_partition,
                        target_cells_per_partition, partition_start,
-                       partition_end, outi, f_view, gf_view, barry_r,
-                       measurements_view, cell_links);
+                       partition_end, outi, f_view, gf_view, f_backup_view,
+                       gf_backup_view, adjc_backup_view, adjv_backup_view,
+                       backup_mutex, barry_r, measurements_view, cell_links);
 }
 
 }  // namespace kernels
 
 clusterization_algorithm::clusterization_algorithm(
     const traccc::memory_resource& mr, vecmem::copy& copy, stream& str,
-    const unsigned short target_cells_per_partition)
+    const config_type& config)
     : m_mr(mr),
       m_copy(copy),
       m_stream(str),
-      m_target_cells_per_partition(target_cells_per_partition) {}
+      m_config(config),
+      m_f_backup(
+          m_config.backup_partition_size * m_config.target_cells_per_partition,
+          m_mr.main),
+      m_gf_backup(
+          m_config.backup_partition_size * m_config.target_cells_per_partition,
+          m_mr.main),
+      m_adjc_backup(
+          m_config.backup_partition_size * m_config.target_cells_per_partition,
+          m_mr.main),
+      m_adjv_backup(m_config.backup_partition_size * 8 *
+                        m_config.target_cells_per_partition,
+                    m_mr.main),
```

Comment:
Could the config struct provide these values with some helper functions? It's a pity to copy-paste these expressions between the CUDA and SYCL code. :thinking: